File: fontes/senha.py
# Teste para uso de senha
# Programa didático, não pode ser
# usado profissionalmente

T = 3
CORRETO = False
while True: # simulação "repeat"
  SENHA = input("Entre a senha de acesso: ")
  if (SENHA == "123"):
    CORRETO = True
  else:
    T = T - 1
    print(f"Tentativas restantes: {T}")
  if (CORRETO == True) or (T == 0): 
    break # simulação "until"
if (CORRETO == True):
  print("Acesso liberado")
else:
  print("Acesso bloqueado")

File: fontes/senha.cs
// Teste para uso de senha
// Programa didático, não pode ser
// usado profissionalmente
// 
// Linguagem C#

using System;

class Program
{
  static void Main()
  {
        
    int T = 3;
    bool CORRETO = false;
    int SENHA;

    do // simulação "repeat"
    {
        Console.Write("Entre a senha de acesso: ");
        string input = Console.ReadLine();
        if (int.TryParse(input, out SENHA))
        {
            if (SENHA == 123)
            {
              CORRETO = true;
            }
            else
            {
              T = T - 1;
              Console.WriteLine("Tentativas restantes: " + T);
            }
        }
        else
        {
          Console.WriteLine("Entrada inválida! Tente novamente.");
        }
    }
    while (!CORRETO == true || T == 0); // simulação "until"
    if (CORRETO == true)
    {
      Console.WriteLine("Acesso liberado");
    }
    else
    {
      Console.WriteLine("Acesso bloqueado");
    }
      
  }
}
